Hackish; needs more work before submitting upstream.

--- origsrc/tomboy-1.0.1/Tomboy/Makefile.am	2009-11-15 23:18:27.000000000 -0600
+++ src/tomboy-1.0.1/Tomboy/Makefile.am	2010-01-31 00:07:28.810004000 -0600
@@ -4,7 +4,9 @@ SUBDIRS = . Addins
 
 TARGET = Tomboy.exe
 WRAPPER = tomboy
+if FIXED_PANELAPPLET
 PANEL_WRAPPER = tomboy-panel
+endif
 
 DBUS_CSFILES = 				\
 	$(srcdir)/IRemoteControl.cs	\
@@ -45,7 +47,6 @@ WIN_LIBS =					\
 endif
 
 GNOME_CSFILES =					\
-	$(srcdir)/Applet.cs 			\
 	$(srcdir)/GConfPreferencesClient.cs	\
 	$(srcdir)/GnomeApplication.cs		\
 	$(srcdir)/GnomeFactory.cs		\
@@ -60,9 +61,7 @@ GNOME_USED_CSFILES = $(GNOME_CSFILES)
 
 if FIXED_PANELAPPLET
 FIXED_PANELAPPLET_CSFLAGS = -define:FIXED_PANELAPPLET
-else
-PANELAPPLET_USED_CSFILES =			\
-	$(PANELAPPLET_CSFILES)
+PANELAPPLET_USED_CSFILES = $(srcdir)/Applet.cs
 endif
 endif
 
--- origsrc/tomboy-1.0.1/Tomboy/Tomboy.cs	2009-11-16 18:00:18.000000000 -0600
+++ src/tomboy-1.0.1/Tomboy/Tomboy.cs	2010-01-30 22:33:58.327102700 -0600
@@ -6,8 +6,6 @@ using Mono.Unix;
 
 #if FIXED_PANELAPPLET
 using Gnome;
-#elif !WIN32 && !MAC
-using _Gnome;
 #endif
 
 using Tomboy.Sync;
@@ -152,7 +150,7 @@ namespace Tomboy
 		static void RegisterPanelAppletFactory ()
 		{
 			// This will block if there is no existing instance running
-#if !WIN32 && !MAC
+#if FIXED_PANELAPPLET
 			PanelAppletFactory.Register (typeof (TomboyApplet));
 #endif
 		}
@@ -641,9 +639,11 @@ namespace Tomboy
 					break;
 #endif // ENABLE_DBUS || WIN32
 
+#if FIXED_PANELAPPLET
 				case "--panel-applet":
 					panel_applet = true;
 					break;
+#endif
 
 				case "--note-path":
 					if (idx + 1 >= args.Length ||
--- origsrc/tomboy-1.0.1/configure.in	2009-11-16 18:00:18.000000000 -0600
+++ src/tomboy-1.0.1/configure.in	2010-01-30 22:48:29.131909900 -0600
@@ -177,8 +177,12 @@ if test "x$ENABLE_GNOME" != "xno"; then
 			          gnome-sharp-2.0
 			          gconf-sharp-2.0
 			          gconf-sharp-peditors-2.0
-			          gnome-panel-sharp-2.24)
-		has_fixed_panelapplet="yes"
+			          gnome-panel-sharp-2.24,
+		has_fixed_panelapplet="yes",
+		[PKG_CHECK_MODULES(GNOME,
+			          gnome-sharp-2.0
+			          gconf-sharp-2.0
+			          gconf-sharp-peditors-2.0)])
 
 	else
 		if pkg-config --atleast-version=2.16.1 gconf-sharp-2.0; then
--- origsrc/tomboy-1.0.1/data/Makefile.am	2009-04-27 12:29:07.000000000 -0500
+++ src/tomboy-1.0.1/data/Makefile.am	2010-01-31 00:07:50.519245700 -0600
@@ -40,9 +40,11 @@ $(dbusservice_DATA): $(dbusservice_in_fi
 endif
 
 
+if FIXED_PANELAPPLET
 serverdir       = $(PANELAPPLET_SERVER_DIR)
 server_in_files = GNOME_TomboyApplet.server.in
 server_DATA     = $(server_in_files:.server.in=.server)
+endif
 
 $(server_in_files): $(server_in_files:.server.in=.server.in.in) Makefile
 	sed -e "s|\@bindir\@|$(bindir)|g"	\
