--- origsrc/tomboy-0.8.2/Tomboy/Addins/PrintNotes/Makefile.am	2007-10-15 09:20:18.000000000 -0500
+++ src/tomboy-0.8.2/Tomboy/Addins/PrintNotes/Makefile.am	2007-12-10 07:19:51.750000000 -0600
@@ -56,7 +56,7 @@
 	gtksourceprintjob.h					\
 	gtksourceprintjob.c
 
-libprintnotes_la_LDFLAGS = -export-dynamic -module -avoid-version
+libprintnotes_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 libprintnotes_la_LIBADD = $(LIBTOMBOY_LIBS)
 
 maintainer-clean-local:
--- origsrc/tomboy-0.8.2/Tomboy/Makefile.am	2007-10-15 09:20:19.000000000 -0500
+++ src/tomboy-0.8.2/Tomboy/Makefile.am	2007-12-10 07:19:51.843750000 -0600
@@ -124,6 +124,10 @@
 
 $(TARGET).config: $(srcdir)/$(TARGET).config.in Makefile
 	sed -e "s|\@pkglibdir\@|$(pkglibdir)|" \
+		-e "s|\@LIB_PREFIX\@|$(LIB_PREFIX)|g" \
+		-e "s|\@LIB_MIDDLE\@|$(LIB_MIDDLE)|g" \
+		-e "s|\@LIB_SUFFIX\@|$(LIB_SUFFIX)|g" \
+		-e "s|\@MOD_SUFFIX\@|$(MOD_SUFFIX)|g" \
 	    < $< > $@
 
 Defines.cs: $(srcdir)/Defines.cs.in Makefile
--- origsrc/tomboy-0.8.2/Tomboy/Tomboy.exe.config.in	2007-10-15 09:20:19.000000000 -0500
+++ src/tomboy-0.8.2/Tomboy/Tomboy.exe.config.in	2007-12-10 07:19:51.859375000 -0600
@@ -1,9 +1,9 @@
 <configuration>
-  <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0.so.0"/>
-  <dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0.so.0"/>	
-  <dllmap dll="libgtkspell" target="libgtkspell.so.0"/>
-  <dllmap dll="libtomboy" target="libtomboy.so"/>
-  <dllmap dll="panel-applet-2" target="libpanel-applet-2.so.0"/>
-  <dllmap dll="dbus-1" target="libdbus-1.so.@DBUS_SOVERSION@"/>
-  <dllmap dll="dbus-glib-1" target="libdbus-glib-1.so.2"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgtk-win32-2.0-0.dll" target="@LIB_PREFIX@gtk-x11-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>	
+  <dllmap dll="libgtkspell" target="@LIB_PREFIX@gtkspell@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgalago-gtk" target="@LIB_PREFIX@galago-gtk@LIB_MIDDLE@1@LIB_SUFFIX@"/>
+  <dllmap dll="libtomboy" target="@LIB_PREFIX@tomboy@MOD_SUFFIX@"/>
+  <dllmap dll="libprintnotes" target="@LIB_PREFIX@printnotes@MOD_SUFFIX@"/>
+  <dllmap dll="panel-applet-2" target="@LIB_PREFIX@panel-applet-2@LIB_MIDDLE@0@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/tomboy-0.8.2/configure.in	2007-11-09 11:12:56.000000000 -0600
+++ src/tomboy-0.8.2/configure.in	2007-12-10 07:19:51.984375000 -0600
@@ -40,6 +40,31 @@
 	AC_MSG_ERROR([Can't find "gmcs" in your PATH])
 fi
 
+case "$host_os" in
+cygwin*)
+	LIB_PREFIX=cyg
+	LIB_MIDDLE=-
+	LIB_SUFFIX=.dll
+	MOD_SUFFIX=.dll
+	;;
+darwin*)
+	LIB_PREFIX=lib
+	LIB_MIDDLE=.
+	LIB_SUFFIX=.dylib
+	MOD_SUFFIX=.so
+	;;
+*)
+	LIB_PREFIX=lib
+	LIB_MIDDLE=.so.
+	LIB_SUFFIX=
+	MOD_SUFFIX=.so
+	;;
+esac
+AC_SUBST(LIB_PREFIX)
+AC_SUBST(LIB_MIDDLE)
+AC_SUBST(LIB_SUFFIX)
+AC_SUBST(MOD_SUFFIX)
+
 #
 # Gdk and Atk are needed for libtrayicon
 #
@@ -155,7 +180,7 @@
 #
 # NUnit required for (optional) unit tests
 #
-PKG_CHECK_MODULES(NUNIT, nunit, HAVE_NUNIT="yes", HAVE_NUNIT="no")
+PKG_CHECK_MODULES(NUNIT, mono-nunit, HAVE_NUNIT="yes", HAVE_NUNIT="no")
 AM_CONDITIONAL(HAVE_NUNIT, test "$HAVE_NUNIT" = "yes")
 AC_SUBST(NUNIT_LIBS)
 AC_PATH_PROG(NUNIT, nunit-console)
--- origsrc/tomboy-0.8.2/libtomboy/Makefile.am	2007-10-15 09:20:15.000000000 -0500
+++ src/tomboy-0.8.2/libtomboy/Makefile.am	2007-12-10 07:19:52.078125000 -0600
@@ -21,7 +21,7 @@
 	eggtrayicon.h		\
 	eggtrayicon.c
 
-libtomboy_la_LDFLAGS = -export-dynamic -module -avoid-version
+libtomboy_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 libtomboy_la_LIBADD = $(LIBTOMBOY_LIBS)
 
 maintainer-clean-local:
